core/src/main/java/com/es/core/model/phone/dao/impl/JdbcPhoneDao.java

```diff
@@ -67,9 +84,55 @@ public void save(final Phone phone) {
     }
 
     @Override
-    @Transactional
     public List<Phone> findAll(int offset, int limit) {
-        return jdbcTemplate.query(SELECT_PHONES_WITH_COLORS, phoneListResultSetExtractor, limit, offset);
+        return jdbcTemplate.query(SELECT_PHONES_WITH_COLORS_WITH_LIMIT_AND_OFFSET, phoneListResultSetExtractor,
+                limit, offset);
+    }
+
+    @Override
+    public List<Phone> findAllInStock(int offset, int limit) {
+        return jdbcTemplate.query(SELECT_PHONES_WITH_COLORS, phoneListResultSetExtractor).stream()
+                .filter(this::isPositivePhoneStock)
+                .skip(offset)
+                .limit(limit)
+                .collect(Collectors.toList());
+    }
+
+    @Override
+    public List<Phone> findAllInStock(String query, SortField sortField, SortOrder sortOrder,
+                                      int offset, int limit) {
+        return jdbcTemplate.query(SELECT_PHONES_WITH_COLORS, phoneListResultSetExtractor).stream()
```

Comment:
This is super ineffective. You fetch whole table with products (almost 8k), convert each of them, and then apply sort and filter for all of them... Just to show one page with 10 phones.
Use SQL for filtering, sorting and pagination